@vates/nbd-client/index.mjs

```diff
@@ -0,0 +1,273 @@
+import assert from 'node:assert'
+import { Socket } from 'node:net'
+import { connect } from 'node:tls'
+import {
+  INIT_PASSWD,
+  MAX_BUFFER_LENGTH,
+  NBD_CMD_READ,
+  NBD_DEFAULT_BLOCK_SIZE,
+  NBD_DEFAULT_PORT,
+  NBD_FLAG_FIXED_NEWSTYLE,
+  NBD_FLAG_HAS_FLAGS,
+  NBD_OPT_EXPORT_NAME,
+  NBD_OPT_REPLY_MAGIC,
+  NBD_OPT_STARTTLS,
+  NBD_REPLY_MAGIC,
+  NBD_REQUEST_MAGIC,
+  OPTS_MAGIC,
+} from './constants.mjs'
+
+// documentation is here : https://github.com/NetworkBlockDevice/nbd/blob/master/doc/proto.md
+
+export default class NbdClient {
+  #serverAddress
+  #serverCert
+  #serverPort
+  #serverSocket
+  #useSecureConnection = false
+
+  #exportname
+  #exportSize
+
+  #receptionBuffer = Buffer.alloc(0)
+
+  // ensure the read are handled in the right order
+  // this is a FIFO
+  #rawReadBacklog = []
+
+  // AFAIK, there is no guaranty the server answers in the same order as the queries
+  #nextCommandQueryId = BigInt(0)
```

Comment:
You could also use `0n`, but your code is ok as it is :slightly_smiling_face:

Reply:
I love explicit and search code, especially on rarely used features